packages/nimble-components/src/anchor-button/tests/anchor-button-matrix.stories.ts

```diff
@@ -17,7 +18,15 @@ import {
 } from '../../utilities/tests/storybook';
 import { hiddenWrapper } from '../../utilities/tests/hidden';
 import { textCustomizationWrapper } from '../../utilities/tests/text-customization';
-import '../../all-components';
+import { AnchorButton } from '..';
+import { IconLink } from '../../icons/link';
+import { IconArrowExpanderRight } from '../../icons/arrow-expander-right';
+
+const nimbleAnchorButton = DesignSystem.tagFor(AnchorButton);
+const nimbleIconLink = DesignSystem.tagFor(IconLink);
+const nimbleIconArrowExpanderRight = DesignSystem.tagFor(
+    IconArrowExpanderRight
+);
```

Comment:
@jattasNI maybe we should just export a `const nimbleAnchorButton = DesignSystem.tagFor(ComponentClass);` from each component?

The `DesignSystem.tagFor(class)` pattern isn't really sticking around for the 

Then can just: `import { nimbleAnchorButton } from '..';`

Name options:
- nimbleAnchorButton (my pref since aligns with the real name well)
- nimbleAnchorButtonTag
- nimbleAnchorButtonTagName
- anchorButtonTag
- anchorButtonTagName

The copy and pasted formatting has a local for that:

```ts
const nimbleAnchorButton = AnchorButton.compose<AnchorOptions>({
    baseName: 'anchor-button',
    template,
    styles,
    shadowOptions: {
        delegatesFocus: true
    }
});

DesignSystem.getOrCreate().withPrefix('nimble').register(nimbleAnchorButton());

```

Which would update the pattern to:
```ts
const anchorButton = AnchorButton.compose<AnchorOptions>({
    baseName: 'anchor-button',
    template,
    styles,
    shadowOptions: {
        delegatesFocus: true
    }
});

DesignSystem.getOrCreate().withPrefix('nimble').register(anchorButton());
export const nimbleAnchorButton = DesignSystem.tagFor(AnchorButton);
```

Reply:
I like the idea of elements exporting a `const` for their tag. I'd vote for its name to include "name", "tag", or both to more clearly distinguish it from existing exports like the class name. And while including "nimble" in the name is kinda nice because it makes it read more like the actual tag name, it's not really necessary and adds extra length. I'd vote for one of `anchorButtonTag` or `anchorButtonName`.

Some other potential work involved in going this direction:
- update icon generation to ensure we export names for each of them
- re-export tag names from nimble-angular so clients don't have to depend on both it and nimble-components
- update existing usages of `DesignSystem.tagFor()` [in client code](https://dev.azure.com/ni/DevCentral/_search?action=contents&text=designsystem.tagfor&type=code&lp=code-Project&filters=ProjectFilters%7BDevCentral%7DRepositoryFilters%7BSkyline%7D&pageSize=25&result=DefaultCollection/DevCentral/Skyline/GBmaster//Web/Workspaces/SystemLinkShared/projects/systemlink-lib-angular/src/sl-dialog/sl-dialog-service.ts) to use the new pattern

Any of those steps which aren't completed as part of this should be tracked in a follow up issue.

Reply:
@fredvisser @rajsite, Fred's initial pass at this pattern looks great to me. 🚀

Reply:
+1 I think the pattern is ready to expand to the other nimble-components. @fredvisser 🚀

Reply:
@rajsite @jattasNI - Should be good to review now. Apologies for 105 files, but you get what you get…

Reply:
Looks like there are quite a few uses of `DesignSystem.tagFor()` in code and a couple references to it in CONTRIBUTING: [github search](https://github.com/search?q=repo%3Ani%2Fnimble%20designsystem.tagfor&type=code)

Could you either update these to use the new API now, do it in a follow up PR, or track it in a follow up issue (along with updating client code)?